[DO NOT MERGE] DAOS-XXXX tests: Check onsite tests so far run fine with release/2.0 branch

saurabhtandan opened the pull request:
PR to verify if tests under ftest/deployment and ftest/performance run without any issues on release/2.0 branch

Test-tag: deployment performance

saurabhtandan committed:
DAOS-7525 test: Adding basic checkout test for installation phase (#6973)

Adding basic_checkout test for verifying basic daos functionality.
Test covers following test cases:
- IO with various different middleware
- Various configuration of metadata generation
- Run daos autotest
- Move data from daos to another daos cont to external POSIX FS and back to daos.
- ior_test_base, mdtest_test_base and datamover_test_base modified and couple of
methods were added for code sharing.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

saurabhtandan committed:
DAOS-7526 test: Adding io & system admin test (#7543)

Adding io & basic system admin test to the deployment
directory to check happy path io tests and system admin
activities.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

saurabhtandan committed:
DAOS-7524 test: Adding test to perform critical integration tasks for deployment cluster (#7728)

Adding test to check for passwordlessssh, version check, ras event
netowrk and storage scan checks on daos server nodes in deployment
cluster.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

daltonbohning committed:
DAOS-8310 test: add ior performance tests (#7803)

- Add ftest/performance with performance tests
  - performance test base
  - IOR Easy DFS SX
  - IOR Easy DFS EC_16P2GX
  - IOR Easy Dfuse SX
  - IOR Easy Dfuse EC_16P2GX
  - IOR Easy DFS EC_4P2GX, stop rank during write
  - IOR Easy DFS EC_4P2GX, stop rank during read
  - IOR Easy DFS EC_16P2GX, stop rank during write
  - IOR Easy DFS EC_16P2GX, stop rank during read
  - IOR Easy HDF5 SX
  - IOR Easy MPIIO SX
  - IOR Hard DFS SX
  - IOR Hard DFS EC_16P2GX
  - IOR Hard Dfuse SX
  - IOR Hard Dfuse EC_16P2GX

- Add IorCmd.sw_deadline
- Adjust IorCmd.get_ior_metrics to accept string input

- Add oclass_utils.py

- Add DaosServerManager.get_daos_metrics
  - Calls daos_metrics for each engine.

- Add SubprocessManager.get_expected_states
  - Returns the expected states for ranks

- Add DaosServerManager.stop_random_rank
  - Stops a random running rank

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

daltonbohning committed:
DAOS-8310 test: add mdtest performance tests (#7959)

- Add mdtest performance tests
  - MDTest Easy DFS S1
  - MDTest Easy DFS EC_16P2G1
  - MDTest Easy DFS EC_4P2G1, stop rank
  - MDTest Easy DFS EC_16P2G1, stop rank
  - MDTest Hard DFS S1
  - MDTest Hard DFS EC_16P2G1

- Add mdtest_test_base.subprocess support

- Add mdtest_utils.MdtestMetrics

Test-tag: basiccheckout

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:329](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/ior_test_base.py#L329):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1157/1000)

[src/tests/ftest/util/server_utils.py:617](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/server_utils.py#L617):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:329](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/ior_test_base.py#L329):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1157/1000)

[src/tests/ftest/util/server_utils.py:617](https://github.com/daos-stack/daos/blob/6161ed96e0b826649296fcf013607b9804caa4a4/src/tests/ftest/util/server_utils.py#L617):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:329](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/ior_test_base.py#L329):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1157/1000)

[src/tests/ftest/util/server_utils.py:617](https://github.com/daos-stack/daos/blob/bd9f925bb9dd5b8b95173ed14f8da305603175de/src/tests/ftest/util/server_utils.py#L617):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

saurabhtandan committed:
Merge branch 'release/2.0' into 2.0.2+onsite_tests

Test-tag: basiccheckout performance

daosbuild1 commented on src/tests/ftest/util/data_mover_test_base.py:
```diff
@@ -1189,3 +1191,101 @@ def run_datamover(self, test_desc=None,
+            read_back_pool = pool
+        # the result is that a NEW directory is created in the destination
+        if tool == 'FS_COPY':
+            daos_path = "/" + os.path.basename(posix_path) + self.ior_cmd.test_file.value
```
(pylint-undefined-variable) Undefined variable 'os'

daosbuild1 replied:
(pylint-undefined-variable) Undefined variable 'os'

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8487/4/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:175](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L175):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1291/1000)

[src/tests/ftest/util/data_mover_test_base.py:216](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L216):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:22](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L22):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/4/execution/node/112/log

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/4/execution/node/363/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/4/execution/node/326/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/4/execution/node/378/log

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8487/5/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:175](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L175):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1291/1000)

[src/tests/ftest/util/data_mover_test_base.py:216](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L216):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:22](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/data_mover_test_base.py#L22):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/b5057639ce754c2cdc1ab5afd24efb0b529ec54e/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/5/execution/node/109/log

saurabhtandan committed:
Fixed missing package import

Test-tag: basiccheckout performance

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/6/execution/node/364/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/6/execution/node/378/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/7/execution/node/374/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/0f30989d0a7098bf549bfec3c9d5dc2151e78d16/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

saurabhtandan committed:
Merge branch 'release/2.0' into 2.0.2+onsite_tests

saurabhtandan committed:
Adding agent_faliure tests frommaster PR-8368 (not merged in master yet)

Test-tag: agent_failure

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/command_utils.py:209](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/command_utils.py#L209):
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'allow_output_check' in constructor call

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/afa1fcf4e8320980bdda2617e2140a9e5e961fb6/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/9/execution/node/371/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/9/execution/node/325/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/9/execution/node/377/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/9/execution/node/374/log

shimizukko committed:
DAOS-9385 test: Add agent failure test (#8368)

Verify agent failure is properly handled.

1. Create a pool and a container.
2. Run IOR.
3. Stop daos_agent process while IOR is running.
4. Check the error on the client side. When daos_agent is killed in the middle of IOR, the file write completes successfully, but the pool disconnect at the end fails, so we get the error message that includes -1005. This step is more like a verification of the test itself rather than the product.
5. Verify daos_server doesn’t have the pool handle opened. Call:
dmg telemetry metrics query --metrics=engine_pool_pool_handles
that shows the number of open pool handle. Verify that it's 0 after the agent is stopped and IOR is finished. This step verifies the isolation of the failure. In this case, daos_agent was stopped, but daos_server can detect it and close the pool handle.
6. Verify journalctl shows the log that the agent is stopped. Call:
journalctl --system -t daos_agent --since <before> --until <after>
This step verifies that DAOS, or daos_agent process in this case, prints useful logs for the user to troubleshoot the issue, which in this case the application can’t be used.
7. Restart daos_agent.
8. Run IOR again. It should succeed this time without any error. This step verifies that DAOS can recover from the fault with minimal human intervention.

Test-tag: basiccheckout criticalintegration agent_failure performance,-manual pr

Signed-off-by: Makito Kano <makito.kano@intel.com>

saurabhtandan committed:
Remove agent_failure tests which were added manually.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/command_utils.py:209](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/command_utils.py#L209):
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'allow_output_check' in constructor call

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/06d7c7ab669d587ce586c00db1553790e7e0894f/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/10/execution/node/354/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/10/execution/node/316/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/10/execution/node/377/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/10/execution/node/369/log

NiuYawei committed:
DAOS-10289 bio: tuneable SPDK subsystem fini timeout

To analyze the issue of SPDK subsystem fini timeout on certain
cluster node, this patch makes the SPDK subsystem fini timeout
tuneable through through env var "DAOS_SPDK_SUBSYS_TIMEOUT".

The default timeout value is 9000 ms, one can change the default
value by setting "DAOS_SPDK_SUBSYS_TIMEOUT" to a different value
or disable timeout by setting "DAOS_SPDK_SUBSYS_TIMEOUT" to 0.

Skip-test: true
Signed-off-by: Niu Yawei <yawei.niu@intel.com>

daosbuild1 commented on src/bio/bio_xstream.c:
```diff
@@ -176,6 +178,14 @@ bio_nvme_init(const char *nvme_conf, int shm_id, int mem_size,
+	/* Hugepages disabled */
+	if (mem_size == 0) {
+		D_INFO("Set per-xstream DMA buffer upper bound to %u %uMB chunks\n",
+			bio_chk_cnt_max, size_mb);
```
```suggestion
		       bio_chk_cnt_max, size_mb);
```

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8487/11/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/command_utils.py:209](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/command_utils.py#L209):
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'allow_output_check' in constructor call

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/a50a7dd93291435a879941fee67a2847302db9e4/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/11/execution/node/129/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/11/execution/node/342/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/11/execution/node/378/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/11/execution/node/353/log

daosbuild1 commented:
Test stage Build on Leap 15 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/11/execution/node/407/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/command_utils.py:209](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/command_utils.py#L209):
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'allow_output_check' in constructor call

[src/tests/ftest/util/data_mover_test_base.py:177](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/data_mover_test_base.py#L177):
(pylint-line-too-long) Line too long (109/100)

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1293/1000)

[src/tests/ftest/util/data_mover_test_base.py:218](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/data_mover_test_base.py#L218):
(pylint-singleton-comparison) Comparison 'path_list == None' should be 'path_list is None'

[src/tests/ftest/util/data_mover_test_base.py:24](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/data_mover_test_base.py#L24):
(pylint-too-many-public-methods) Too many public methods (21/20)

[src/tests/ftest/util/ior_test_base.py:315](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/ior_test_base.py#L315):
(pylint-inconsistent-return-statements) Either all return statements in a function should return an expression, or none of them should.

[src/tests/ftest/util/server_utils.py:1](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/server_utils.py#L1):
(pylint-too-many-lines) Too many lines in module (1158/1000)

[src/tests/ftest/util/server_utils.py:618](https://github.com/daos-stack/daos/blob/80b91b35b22cd34cdf4b227d6272b3652bca51d4/src/tests/ftest/util/server_utils.py#L618):
(pylint-consider-using-set-comprehension) Consider using a set comprehension
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/12/execution/node/380/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/12/execution/node/314/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/12/execution/node/406/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-8487/12/execution/node/394/log
